test/tests/functional/pbs_qstat_json.py

```diff
@@ -0,0 +1,292 @@
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# The PBS Pro software is licensed under the terms of the GNU Affero General
+# Public License agreement ("AGPL"), except where a separate commercial license
+# agreement for PBS Pro version 14 or later has been executed in writing with
+# Altair.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software - under
+# a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from ptl.utils.pbs_testsuite import *
+import json
+
+
+@tags('smoke')
+class Qstat_json(PBSTestSuite):
+    """
+    Test qstat output can be viewed in json format
+    """
+
+    def parse_json(self, dict, qstat_attr):
+        """
+        Common function for parsing the all values in json output
+        """
+        for key, val in dict.items():
+            qstat_attr.append(str(key))
+            if type(val) == type(dict):
+                for key, val in val.items():
+                    qstat_attr.append(str(key))
+                    if type(val) == type(dict):
+                        self.parse_json(val, qstat_attr)
+        return qstat_attr
+
+    def test_qstat_json_valid(self):
+        """
+        Test json output of qstat -f is in valid format when querired as a
+        super user and all attributes displayed in qstat are present in output
+        """
+        j = Job(TEST_USER)
+        j.set_sleep_time(10)
+        jid = self.server.submit(j)
+        f = open('/tmp/qstat_script', 'w')
+        f.write(self.server.pbs_conf['PBS_EXEC'] +
+                "/bin/qstat -f -F json > /tmp/qstat_out")
+        f.close()
+        os.chmod("/tmp/qstat_script", 0755)
+        run_script = "sh /tmp/qstat_script"
+        ret = self.du.run_cmd(self.server.hostname, cmd=run_script, sudo=True)
+        print ret['out']
+        f = open('/tmp/qstat_out', 'r')
+        qstat_out = f.read()
```

Comment:
ret['out'] breaks json output line by line and makes it an array of lines . Which makes it unusable for json modules.